specification/under_development/profiles/Base_Profile.md

```diff
@@ -0,0 +1,484 @@
+  tail call void @__quantum__qis__mz__body(%Qubit* null, writeonly %Result* null)
+  tail call void @__quantum__qis__mz__body(%Qubit* inttoptr (i64 1 to %Qubit*), writeonly %Result* inttoptr (i64 1 to %Result*))
+  br label %output
+
+output:                                   ; preds = %entry
+  ; calls to record the program output
+  tail call void @__quantum__rt__tuple_record_output(i64 2, i8* null)
+  tail call void @__quantum__rt__result_record_output(%Result* null, i8* getelementptr inbounds ([3 x i8], [3 x i8]* @0, i32 0, i32 0))
+  tail call void @__quantum__rt__result_record_output(%Result* inttoptr (i64 1 to %Result*), i8* getelementptr inbounds ([3 x i8], [3 x i8]* @1, i32 0, i32 0))
+
+  ret i64 0
+}
+
+; declarations of QIS functions
+
+declare void @__quantum__qis__h__body(%Qubit*)
+
+declare void @__quantum__qis__cnot__body(%Qubit*, %Qubit*)
+
+declare void @__quantum__qis__mz__body(%Qubit*, writeonly %Result*)
+
+; declarations of functions used for output recording
+
+declare void @__quantum__rt__tuple_record_output(i64, i8*)
+
+declare void @__quantum__rt__result_record_output(%Result*, i8*)
+
+; attributes
+
+attributes #0 = { "entry_point" "qir_profile"="base_profile" "output_labels"="schema_id" "required_qubits"="2" "required_results"="2" }
+
+; module flags
+
+!llvm.module.flags = !{!0, !1, !2, !3}
+
+!0 = !{i32 1, !"qir_major_version", i32 1}
+!1 = !{i32 7, !"qir_minor_version", i32 0}
+!2 = !{i32 1, !"dynamic_qubit_management", i1 false}
+!3 = !{i32 1, !"dynamic_result_management", i1 false}
+```
+
+The program entangles two qubits, measures them, and returns a tuple with the
+two measurement results.
+
+For the sake of clarity, the code above does not contain any [debug
+symbols](https://releases.llvm.org/13.0.0/docs/tutorial/MyFirstLanguageFrontend/LangImpl09.html?highlight=debug%20symbols).
+Debug symbols contain information that is used by a debugger to relate failures
+during execution back to the original source code. While we expect this to be
+primarily useful for execution on simulators, debug symbols are both easy to
+ignore and may be useful to generate helpful error messages for compilation
+failures that happen only late in the process. We hence see no reason to
+disallow them from occurring in the bitcode but will not detail their use any
+further. We defer to existing resources for more information about how to
+generate and use debug symbols.
+
+## Entry Point Definition
+
+The bitcode contains the definition of the LLVM function that should be invoked
+when the program is executed, referred to as entry point in the rest of this
+profile specification. The name of this function may be chosen freely, as long
+as it is a valid [global
+identifier](https://llvm.org/docs/LangRef.html#identifiers) by LLVM standard.
+
+The entry point may not take any parameters and must return an exit code in the
+form of a 64-bit integer. The exit code `0` must be used to indicate a
+successful execution of the program.
```

Comment:
What if execution is not succesful? Could it return anything else anyhow? Not being allowed to branch and QIS functions not returning anything, it will be hard to set a return value. Unless we use tricky workarounds (see [this discussion](https://github.com/qir-alliance/qir-spec/discussions/15)). Maybe we should say it always return 0?

Reply:
As you point out, I don't see how the Base Profile could return anything but 0. The intent to return it regardless is to be consistent with other profiles that can return a meaningful exit code.

Reply:
Is there not any case in which the Base Profile execution would return an error code?  Are you thinking that all errors would be captured in compilation phase? what if the code fails on a provider's system because they have not properly implemented some feature and the control system exits prematurely.  Seems like there must be some cases where a failure code could surface, not from the QIS code, but from the control system itself?

Reply:
@rtvuser1 I agree, but I don't quite see how that could or should be reflected in the IR. We could consider adding a sentence that a backend may choose to replace the program defined exit code if there was an execution failure, but I am not sure it is the right thing to do; maybe this should not be covered by the IR spec, but instead part of a service/job API?